### PR TITLE
ioctl bc bucket count/max(#2353)

### PR DESCRIPTION
**Pull request by JonSnow47**
ioctl bc bucket count/max
when iotex-proto merge into master, I'll update `github.com/iotexproject/iotex-proto` version in `go.mod`

### CODE REVIEW

File: action/protocol/staking/read_state.go [hunk 1]
```diff
@@ -101,6 +101,21 @@ func readStateBucketByIndices(ctx context.Context, sr protocol.StateReader,
...
+	total, err := getTotalBucketCount(csr.SR())
+	if errors.Cause(err) == state.ErrStateNotExist {
+		return &iotextypes.BucketsCount{}, csr.Height(), nil
+	}
+	if err != nil {
+		return nil, 0, err
+	}
+	return &iotextypes.BucketsCount{
+		Total:  total,
+		Active: csr.ActiveBucketsCount(),
```
Comment (dustinxie) [inline action/protocol/staking/read_state.go:115]: for active count, similar to how to handle total amount in 
`func getTotalStakedAmount(ctx context.Context, csr CandidateStateReader) (*big.Int, error)`

Comment (JonSnow47) [reply action/protocol/staking/read_state.go:115]: is that means we should have a func like `getActiveBucketsCount(ctx context.Context, csr CandidateStateReader) uint64` to return active count?

File: action/protocol/staking/read_state.go [hunk 2]
```diff
@@ -216,3 +231,7 @@ func getTotalStakedAmount(ctx context.Context, csr CandidateStateReader) (*big.I
 	// otherwise read from bucket pool
 	return csr.TotalStakedAmount(), nil
 }
+
+func getActiveBucketsCount(csr CandidateStateReader) uint64 {
```
Comment (dustinxie) [inline action/protocol/staking/read_state.go:235]: i mean, similar to the code in getTotalStakedAmount()

if after greenland {
     // read state from db
    return total.count
}

return csr.ActiveBucketsCount(), nil

File: action/protocol/staking/read_state.go [hunk 3]
```diff
@@ -232,6 +236,17 @@ func getTotalStakedAmount(ctx context.Context, csr CandidateStateReader) (*big.I
...
 }
 
-func getActiveBucketsCount(csr CandidateStateReader) uint64 {
-	return csr.ActiveBucketsCount()
+func getActiveBucketsCount(ctx context.Context, csr CandidateStateReader) (uint64, error) {
+	chainCtx := protocol.MustGetBlockchainCtx(ctx)
+	hu := config.NewHeightUpgrade(&chainCtx.Genesis)
+	if hu.IsPost(config.Greenland, csr.Height()) {
+		// after Greenland, read state from db
+		buckets, _, err := getAllBuckets(csr.SR())
```
Comment (dustinxie) [inline action/protocol/staking/read_state.go:244]: var total totalAmount
		_, err := csr.SR().State(&total, protocol.NamespaceOption(StakingNameSpace), protocol.KeyOption(bucketPoolAddrKey))

return total.count, nil

File: action/protocol/staking/read_state.go [hunk 4]
```diff
@@ -243,7 +243,7 @@ func getActiveBucketsCount(ctx context.Context, csr CandidateStateReader) (uint6
 		// after Greenland, read state from db
 		var total totalAmount
 		_, err := csr.SR().State(&total, protocol.NamespaceOption(StakingNameSpace), protocol.KeyOption(bucketPoolAddrKey))
-		if err != nil {
+		if err != nil && errors.Cause(err) != state.ErrStateNotExist {
```
Comment (dustinxie) [inline action/protocol/staking/read_state.go:246]: why need this check?
if pass Greenland and state does not exist, it is a real error that should not be escaped

**Pull request merged**